### ISSUE TITLE
[Bug]: Firefox Android doesn't ask to save passwords in apps

### ISSUE THREAD

**Issue by Antorell** [mozilla-mobile/fenix#28547]
### Steps to reproduce

Type a password that is not saved in Firefox's password manager, Firefox doesn't ask to save the password after the login succeeded.

### Expected behaviour

Ask to save the password like other browsers or password managers do when set as default Autofill Provider.

### Actual behaviour

Doesn't to anything. 

### Device name

Redmi Note 11 NFC (EU version)

### Android version

Android 12

### Firefox release type

Firefox Nightly

### Firefox version

110.0a1/108.2

### Device logs

None

### Additional information

**Firefox 110.0a1 set as default Autofill** (Same behavior with the current version of Firefox stable 108.2)
The behavior is the same if I unlock Firefox's Autofill (Fingerprint request) or not. 

https://user-images.githubusercontent.com/25066224/212687552-d5d8e91e-454a-46f6-af15-3dc8ceba901c.mp4

------------------------------

**Chromium Edge for Android set as default autofill** (Same behavior with Microsoft Authenticator):

https://user-images.githubusercontent.com/25066224/212688153-642ed250-c3ca-488a-8b33-e6c717546bb6.mp4

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-22276)

**Comment by Mugurell** [comment]
@Antorell Thanks for opening this!
Just to confirm, in both of the above videos you were using Microsoft Edge as the browser in which you were logging in?

**Comment by Mugurell** [comment]
Thank you!
Can you try the other way around - browsing with Firefox Nightly and using another password manager to see if that other password manager will ask to save passwords types in Firefox?
Firefox will offer to remember password typed when browsing in Firefox but not when typed in other apps. I think Microsoft Edge does the same.

**Comment by Antorell** [comment]
If I set Chromium Edge for Android as default Autofill, no it doesn't ask me to save passwords I enter in Firefox. Firefox's password manager takes over. The same thing happen with Google Chrome, or Chromium Edge if I set another Autofill Provider as default Autofil Provider.  This is the expected behavior as they all have their own password manager. 

This is not the issue I'm reporting. 

I'm reporting that Firefox does not ask to save login credentials in Android applications when it is set as default Autofill provider like other Autofill Providers do. 

Here for example; Chromium Edge for Android when it is set as default Autofill Provider asking to save a new password on the [Casino MAX](https://play.google.com/store/apps/details?id=fr.casino.fidelite&hl=en&gl=US) app:

https://user-images.githubusercontent.com/25066224/212688153-642ed250-c3ca-488a-8b33-e6c717546bb6.mp4
